### DISCUSSION TITLE
How to add a subtitle?

### DISCUSSION THREAD

**Discussion by szgaljic** [mmistakes/minimal-mistakes#3103] [Q&A]
## Description
Is it possible to add a sub-title to my pages? I hacked this by adding a <small> block at the top of my markdown pages but there is still an odd looking gap between the title and the subtitle.

```
---
title:  "Very cool page"
permalink: /cool
---

<small>You're going to love this.</small>

## First Section
Text.

## Second Section
Text.
```

## Environment

- Minimal Mistakes version:
- Ruby gem or remote theme version:
- Jekyll version:
- Git repository URL:
- Operating system:
- GitHub Pages hosted (if yes provide URL to site):

---

**Comment by mmistakes** [comment]
You'll probably need to hack the theme if you want to do this. You could add front matter to your posts like:

```yaml
---
title: "Very cool page"
subtitle: "You're going to love this."
permalink: /cool
---
```

And then add it to whatever layout you're using with:

```liquid
{% if page.subtitle %}{{ page.subtitle }}{% endif %}
```

A good place for that would be right after the [page's title](https://github.com/mmistakes/minimal-mistakes/blob/master/_layouts/single.html#L29) with whatever styling you'd need.

**Comment by szgaljic** [comment]
Okay thanks!

**Comment by amirkdv** [comment]
For anyone else ending up here: you don't need to hack the theme. You can just populate the `page__meta.html` template which is included by the  theme, see [code](https://github.com/mmistakes/minimal-mistakes/blob/9269842e9dceed85021d657bdc80fb0ddbb26b3c/_layouts/single.html#L32).

Example:

```jinja
<!-- _includes/page__meta.html -->

{% if page.subtitle %}
    <p id="page-subtitle" class="page__subtitle p-name" itemprop="headline">
    {{ page.subtitle }}
    </p>
{% endif %}

```

**Reply by ghost** [reply]
This worked perfectly for me with the remote theme. Thanks!

**Comment by safoorsafdar** [comment]
Adding page__meta.html was hiding my time to read from a single post. So instead of following this approach, I added the style to differentiate paragraph and subtitle. 

Add the new css in `assets/css/main.scss`
````css
.page__subtitle{
    border-bottom: 1px solid $light-gray;
    font-size: $type-size-5;
    color: $gray;
    padding-bottom: $type-size-5;
}
````
then in my post:

There is no one who loves pain itself, who seeks after it and wants to have it, simply because it is pain...
{: .page__subtitle}

and in my post its look like this:
![image](https://user-images.githubusercontent.com/5680768/146767763-586bfa7f-8ae5-4209-ac35-802d176a0e30.png)

You can apply your own styles. This way the first paragraph will be part of the excerpt and your post description automatically.

**Reply by barriejc** [reply]
Thank you, this totally worked once I had it in the right place!  Something to note for anyone else stumbling upon this, this css snippet needs to go *after* the @import of minimal-mistakes or the variables will not be set.  

Also, to keep this in line with the skin I was using, I replaced `color: $gray;` with `color: $muted-text-color;` (also $light-gray with $footer-background-color but you may not want the line to be the same color as your footer; check out the skin variables in \_sass/minimal-mistakes/skins/_[skin].scss where [skin] is the name of the one you like or are using).
